realsense2_camera/src/realsense_node_factory.cpp

```diff
@@ -158,10 +144,25 @@ void RealSenseNodeFactory::onInit()
 		}
 		else
 		{
-			std::function<void(rs2::event_information&)> change_device_callback_function = [this](rs2::event_information& info){change_device_callback(info);};
-			_ctx.set_devices_changed_callback(change_device_callback_function);
 			privateNh.param("initial_reset", _initial_reset, false);
 			_device = getDevice();
+			if (_device && _initial_reset)
+			{
+				_initial_reset = false;
+				try
+				{
+					ROS_INFO("Resetting device...");
```

Comment:
If you know the device SN, it might be helpful to include it in the INFO message.

Reply:
The driver does print that out as a INFO message in the rest of the code, during initialization

Reply:
So you will add it or no?

Reply:
Doesn't it become redundant then? Or are you saying the INFO statement here would say something like
```
Looking for device with ID: <Serial> -> Not yet displayed
...
... Later in the initialization process ...
...
Realsense with device ID <Serial> found -> Currently already displayed
```